### PR TITLE
chore: delete old feature flags from DB

### PR DESCRIPTION
**Pull request by frosso**
#### Changes proposed in this Pull Request

I created a migration to delete old feature flags from the DB.
I am creating it because it looks like they might have never been cleaned.
Some of the feature flags might have been internal/developer facing, but others might have been customer-facing (like the UPE ones).

Cleaning them all up.

To gather this list, I went through the history of the `includes/class-wc-payments-features.php` and all its changes.

#### Testing instructions

I added som tests for the migration.

- Insert one of the deprecated options in your DB to an older one - something like:
```sql
INSERT INTO `wp_options` (`option_name`, `option_value`, `autoload`)
VALUES ('_wcpay_feature_client_secret_encryption', '1', 'yes');
```
- Change the value of WooPayments on your DB - something like:
```sql
UPDATE `wp_options` SET
`option_value` = '7.1.1'
WHERE `option_name` = 'woocommerce_woocommerce_payments_version';
```
- Change the local version of WooPayments here to `8.1.0`: https://github.com/Automattic/woocommerce-payments/blob/develop/woocommerce-payments.php#L14-L14
- Load the WooPayments settings page
- In your DB, check that:
  - The WooPayments version changed to `8.1.0`
```sql
SELECT * FROM `wp_options` WHERE `option_name` = 'woocommerce_woocommerce_payments_version' LIMIT 50
```
  - The `_wcpay_feature_client_secret_encryption` option no longer exists
```sql
SELECT * FROM `wp_options` WHERE `option_name` = '_wcpay_feature_client_secret_encryption'
```

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9263` or branch name `chore/delete-old-feature-flags` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=chore/delete-old-feature-flags) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 884581dfd507a94ccf4a9fb3be61f57d51e71eaa
- Build time: 2024-08-15 14:45:25 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/payment-details/summary/test/index.test.tsx [hunk 1]
```diff
@@ -33,7 +33,6 @@ declare const global: {
 		};
 		featureFlags: {
 			isAuthAndCaptureEnabled: boolean;
-			isRefundControlsEnabled: boolean;
```
Comment (frosso) [inline client/payment-details/summary/test/index.test.tsx:36]: This flag was only used in tests - likely missed during a cleanup task.

File: includes/class-wc-payments-features.php [hunk 2]
```diff
@@ -386,7 +367,6 @@ public static function to_array() {
 				'woopayExpressCheckout'          => self::is_woopay_express_checkout_enabled(),
 				'isAuthAndCaptureEnabled'        => self::is_auth_and_capture_enabled(),
 				'isDisputeIssuerEvidenceEnabled' => self::is_dispute_issuer_evidence_enabled(),
-				'isRefundControlsEnabled'        => self::is_streamline_refunds_enabled(),
```
Comment (frosso) [inline includes/class-wc-payments-features.php:389]: This flag was only used in tests - likely missed during a cleanup task.

File: includes/class-wc-payments-features.php [hunk 3]
```diff
@@ -47,15 +46,6 @@ public static function is_tokenized_cart_prb_enabled(): bool {
...
 	}
 
-	/**
-	 * Checks whether streamline refunds is enabled.
-	 *
-	 * @return bool
-	 */
-	public static function is_streamline_refunds_enabled(): bool {
-		return '1' === get_option( self::STREAMLINE_REFUNDS_FLAG_NAME, '0' );
-	}
```
Comment (frosso) [inline includes/class-wc-payments-features.php:57]: This method was only used in tests - likely missed during a cleanup task.

File: includes/class-wc-payments-features.php [hunk 4]
```diff
@@ -188,15 +178,6 @@ public static function maybe_enable_wcpay_subscriptions_after_onboarding( $onboa
...
 	}
 
-	/**
-	 * Returns whether WCPay Subscription migration is enabled
-	 *
-	 * @return bool
-	 */
-	public static function is_subscription_migration_enabled() {
-		return '1' === get_option( '_wcpay_feature_allow_subscription_migrations', '0' );
-	}
```
Comment (frosso) [inline includes/class-wc-payments-features.php:198]: This method was not used anywhere in the codebase.

File: tests/unit/bootstrap.php [hunk 5]
```diff
@@ -54,8 +54,6 @@ function () {
 		}
 	);
 
-	update_option( '_wcpay_feature_allow_subscription_migrations', '1' );
```
Comment (frosso) [inline tests/unit/bootstrap.php:57]: This flag was likely a leftover from a cleanup task

File: tests/unit/test-class-wc-payments-order-service.php [hunk 6]
```diff
@@ -37,16 +37,6 @@ public function set_up() {
 
 		$this->order_service = new WC_Payments_Order_Service( $this->createMock( WC_Payments_API_Client::class ) );
 		$this->order         = WC_Helper_Order::create_order();
-
-		// Turn on fraud settings.
-		update_option( 'wcpay_fraud_protection_settings_active', '1' );
```
Comment (frosso) [inline tests/unit/test-class-wc-payments-order-service.php:42]: This flag was not checked anywhere in the codebase, just in tests. Likely a leftover from a cleanup task.

File: tests/unit/test-class-wc-payments-token-service.php [hunk 7]
```diff
@@ -175,7 +175,6 @@ public function test_add_token_to_user_for_sepa() {
 	 * Test add SEPA token to user with deferred intent creation UPE.
 	 */
 	public function test_add_token_to_user_for_sepa_deferred_intent_creation_upe() {
-		$this->mock_cache->method( 'get' )->willReturn( [ 'is_deferred_intent_creation_upe_enabled' => true ] );
```
Comment (frosso) [inline tests/unit/test-class-wc-payments-token-service.php:178]: This flag was not checked anywhere in the codebase - likely a leftover from a cleanup task.

File: includes/migrations/class-erase-deprecated-flags-and-options.php [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+		delete_option( '_wcpay_feature_client_secret_encryption' );
+		delete_option( '_wcpay_feature_allow_subscription_migrations' );
+		delete_option( '_wcpay_feature_custom_deposit_schedules' );
+		delete_option( '_wcpay_feature_account_overview_task_list' );
+		delete_option( '_wcpay_feature_account_overview' );
+		delete_option( '_wcpay_feature_sepa' );
+		delete_option( '_wcpay_feature_sofort' );
+		delete_option( '_wcpay_feature_giropay' );
+		delete_option( '_wcpay_feature_grouped_settings' );
+		delete_option( '_wcpay_feature_customer_multi_currency' );
```
Comment (gpressutto5) [inline includes/migrations/class-erase-deprecated-flags-and-options.php]: It looks like this flag is still being used.

https://github.com/Automattic/woocommerce-payments/blob/06ede9f56a7e1a0142b5c632a345da36728f8483/includes/class-wc-payments-features.php#L73-L75

https://github.com/Automattic/woocommerce-payments/blob/06ede9f56a7e1a0142b5c632a345da36728f8483/includes/admin/class-wc-rest-payments-settings-controller.php#L750

**Review by gpressutto5 — CHANGES_REQUESTED** [review]

File: includes/migrations/class-erase-deprecated-flags-and-options.php (hunk 8 above)
Comment (frosso) [reply includes/migrations/class-erase-deprecated-flags-and-options.php]: 🤦 I thought I double-checked everything, thank you for catching it!
```suggestion
```

**Review by gpressutto5 — APPROVED** [review]

**Pull request merged**